### PR TITLE
Adding OVN-DPDK blog

### PR DESCRIPTION
**Pull request by leongold**
This blog describes OVN-DPDK support in oVirt.

### CODE REVIEW

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+For more information about OVN localnet integration in oVirt, refer to oVirt's [Provider Physical Network RFE](https://ovirt.org/develop/release-management/features/network/provider-physical-network/).
+
+# Usage in oVirt
+1) Install `oVirt.dpdk-setup` Ansible role via ansible-galaxy:
+    ```
+    # ansible-galaxy install oVirt.dpdk-setup
+    ```
+2) Execute the role as described above.
+3) Unless the host was already configured, the host will restart for kernel changes to be applied.
+4) Make sure the following:
```
Comment (dankenigsberg) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: Make sure *that after reboot*:

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    * IOMMU: ```/sys/devices/virtual/iommu/<DMAR device>/devices/<PCI address>``` should exist. DMAR devices are typically annotated by dmar0, dmar1, and so forth.
+    * hugepages: ```grep Huge /proc/meminfo``` should reflect the desired state as defined in the Ansible playbook (i.e. `Hugepagesize` and `HugePages_Total`)
+    * CPU partitioning: based on the devices that are to be used with a DPDK compatible driver, CPU cores should be partitioned separately on each NUMA node. Refer to `lscpu` to see live CPU NUMA separation information, e.g.: `NUMA node1 CPU(s):     8-15`
```
Comment (dankenigsberg) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: you ask to "make sure", but what should I do if it does not? is it likely to fail?

maybe we'd like to use a different language, such "note that after reboot the following values are changed"
If it does not, you probably have an incompatible hardware or should report a bug on user@ovirt.org.

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+5) Create an oVirt network ("localnet-0"). This network will be used to attach a DPDK NIC.
```
Comment (dankenigsberg) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: Create an oVirt network ("localnet-0") *on your datacenter and attach it to your cluster*.

This network will be used to attach a DPDK NIC -> You should attach this network to DPDK NICs accross your cluster.

Comment (dankenigsberg) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: Please add your beatiful demo screenshots

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+6) Create an external OVN network ("external-net-0"), setting "localnet-0" as its physical network. This configuration between "external-net-0" and "localnet-0" will enable traffic between "localnet-0" (which DPDK's physical port is part of) and the rest of the ports present in OVN's "external-net-0" network.
+7) "external-net-0" vNic's may now be added to virtual machines. These vNics now share L2 connectivity with other remote ports in "external-net-0" via DPDK's NIC (and other local ports internally).
+
+Note:
+* Virtual machines using DPDK based NIC's need to have both `hugepages_shared` (to `true`) and `hugepages` (to the amount of requested shared hugepages) custom properties set. Host to VM packets are transmitted and received on buffers allocated on shared hugepages memory. As of date, these properties are not automated.
```
Comment (dankenigsberg) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: no need for a "note". just add another action item. E.g. for every VM that needs DPDK connectivity, set hugepages_shared...

Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 3 above)
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done

Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 2 above)
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 1 above)
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    - ovirt-ansible-dpdk-setup
+```
+The role is controlled by 3 boolean variables (all set to `true` by default) and a dictionary of devices and their drivers:
+* `configure_kernel` -- determines whether the kernel should be configured for DPDK usage (hugepages, CPU partitioning)
+* `bind_drivers` -- determines whether the role should bind devices to their specified drivers.
+* `set_ovs_dpdk` -- determines whether the role should initialize and configure OVS for DPDK usage.
+* `pci_drivers` -- a dictionary of the PCI addresses of network interface cards as keys and drivers as values. Empty strings represent the kernel default. non-DPDK compatible drivers won't be configured for OVS-DPDK usage.
+
+Additionally, there are several optional performance related variables:
+* `pmd_threads_count` -- determines the amount of PMD threads per NIC (default: `1`).
```
[the remark this answers is not in the document]
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    bind_drivers: true
+    set_ovs_dpdk: false
+  roles:
+    - ovirt-ansible-dpdk-setup
+```
+The role is controlled by 3 boolean variables (all set to `true` by default) and a dictionary of devices and their drivers:
+* `configure_kernel` -- determines whether the kernel should be configured for DPDK usage (hugepages, CPU partitioning)
+* `bind_drivers` -- determines whether the role should bind devices to their specified drivers.
+* `set_ovs_dpdk` -- determines whether the role should initialize and configure OVS for DPDK usage.
+* `pci_drivers` -- a dictionary of the PCI addresses of network interface cards as keys and drivers as values. Empty strings represent the kernel default. non-DPDK compatible drivers won't be configured for OVS-DPDK usage.
```
[the remark this answers is not in the document]
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+      "0000:02:00.2": "igb"
+      "0000:02:00.3": ""
+    configure_kernel: true
+    bind_drivers: true
+    set_ovs_dpdk: false
+  roles:
+    - ovirt-ansible-dpdk-setup
+```
+The role is controlled by 3 boolean variables (all set to `true` by default) and a dictionary of devices and their drivers:
+* `configure_kernel` -- determines whether the kernel should be configured for DPDK usage (hugepages, CPU partitioning)
```
[the remark this answers is not in the document]
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+While still experimental, OVN-DPDK in oVirt is now available in both the stable and master oVirt branches. 
+
+# What's changed?
+
+### Ansible DPDK host setup
+
+Host configuration for DPDK usage is now automated using Ansible. This primarly includes:
+* Hugepages configuration -- hugepage size and quantity in the kernel.
+* CPU partitioning.
+* NIC to userspace driver binding.
```
[the remark this answers is not in the document]
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 9]
```diff
@@ -41,15 +41,15 @@ An example playbook:
...
-* `configure_kernel` -- determines whether the kernel should be configured for DPDK usage (hugepages, CPU partitioning)
+* `configure_kernel` -- determines whether the kernel should be configured for DPDK usage (hugepages, CPU partitioning). WARNING: When set to `true` it is very likely to trigger reboot of the host, unless all required configuration is already pre-set.
 * `bind_drivers` -- determines whether the role should bind devices to their specified drivers.
 * `set_ovs_dpdk` -- determines whether the role should initialize and configure OVS for DPDK usage.
-* `pci_drivers` -- a dictionary of the PCI addresses of network interface cards as keys and drivers as values. Empty strings represent the kernel default. non-DPDK compatible drivers won't be configured for OVS-DPDK usage.
+* `pci_drivers` -- a dictionary of the PCI addresses of network interface cards as keys and drivers as values. Empty strings represent kernel's default. non-DPDK compatible drivers may be set; NICs not using DPDK compatible drivers won't be configured for OVS-DPDK usage and won't have their CPU isolated in multiple NUMA environments.
```
Comment (dankenigsberg) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: still not clear if specifying a bad pci device, we end up with an error code

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 10]
```diff
@@ -65,14 +65,17 @@ For more information about OVN localnet integration in oVirt, refer to oVirt's [
...
-7) "external-net-0" vNic's may now be added to virtual machines. These vNics now share L2 connectivity with other remote ports in "external-net-0" via DPDK's NIC (and other local ports internally).
-
-Note:
-* Virtual machines using DPDK based NIC's need to have both `hugepages_shared` (to `true`) and `hugepages` (to the amount of requested shared hugepages) custom properties set. Host to VM packets are transmitted and received on buffers allocated on shared hugepages memory. As of date, these properties are not automated.
+    * CPU partitioning: based on the devices that are to be used with a DPDK compatible driver, CPUs should be partitioned separately between each NUMA node. Refer to `lscpu` to see live CPU NUMA separation information, e.g.: `NUMA node1 CPU(s):     8-15`
+4) Create an oVirt network ("phys-net-demo") on your data center and attach it to your cluster.
+![](https://image.ibb.co/mm7LJd/5_create_phys_net.png)
```
Comment (dankenigsberg) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: "Also, don't use Imgbb to host image libraries you link to from elsewhere, content for your website, advertising, avatars, or anything else that turns us into your content delivery network."

please upload your images to ovirt-site (you can include them in your git commit)

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 11]
```diff
@@ -65,14 +65,17 @@ For more information about OVN localnet integration in oVirt, refer to oVirt's [
...
+5) Attach "phys-net-demo" to DPDK NICs accross your cluster.
+![](https://image.ibb.co/ihUpBy/7_set_phys_net.png)
+6) Create an external OVN network ("ext-net-demo"), setting "phys-net-demo" as its physical network. This configuration between "ext-net-demo" and "phys-net-demo" will enable traffic between "phys-net-demo" (which DPDK's physical port is part of) and the rest of the ports present in OVN's "ext-net-demo" network.
+![](https://image.ibb.co/iT0J4J/6_create_ext_net.png)
+7) "ext-net-demo" vNic's may now be added to virtual machines. These vNics now share L2 connectivity with other remote ports in "ext-net-demo" via DPDK's NIC (and other local ports internally).
+![](https://image.ibb.co/bBqFJd/ext_net_demo.png)
+8) Virtual machines using DPDK based NIC's need to have both `hugepages_shared` (to `true`) and `hugepages` (to the number of requested shared hugepages) custom properties set. Host to VM packets are transmitted and received on buffers allocated on shared hugepages memory. As of date, these properties are not automated.
```
Comment (dankenigsberg) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: it is hard to review such a single long line. break it to multiple short one. you can do it without ruining markdown paragraphs.

First sentence is way too complex. you can break it to two. or three. How many pages should a typical user set?
Second sentence should probably come first: To transfer packets efficiently between VM and host...
Third sentence: no idea what it means. do we have any plans to automate something? what is it?

does a user need to do someting to enable these custom properties?

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 12]
```diff
@@ -65,14 +65,17 @@ For more information about OVN localnet integration in oVirt, refer to oVirt's [
     # ansible-galaxy install oVirt.dpdk-setup
     ```
 2) Execute the role as described above.
-3) Unless the host was already configured, the host will restart for kernel changes to be applied.
-4) Make sure the following:
+3) Unless the host was already configured, the host will restart for kernel changes to be applied. Note that after reboot the following values are changed (If they are not, you may have incompatible hardware, or you should report a bug on ovirt@users.org):
     * IOMMU: ```/sys/devices/virtual/iommu/<DMAR device>/devices/<PCI address>``` should exist. DMAR devices are typically annotated by dmar0, dmar1, and so forth.
     * hugepages: ```grep Huge /proc/meminfo``` should reflect the desired state as defined in the Ansible playbook (i.e. `Hugepagesize` and `HugePages_Total`)
```
Comment (dankenigsberg) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: please add a word about NICs changing names.

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 10 above)
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: haha, oops! didn't mean to commit with imgbb

Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 11 above)
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+tags: OVS-DPDK, Open vSwitch, OVN, ansible
+date: 2018-07-19 10:00:00 CET
+comments: true
+published: true
+---
+[DPDK](http://dpdk.org/) (Data Plane Development Kit) provides high-performance packet processing libraries and user space drivers.
+
+[Support for DPDK](https://www.ovirt.org/blog/2017/09/ovs-dpdk/) in oVirt was introduced in 2017, and is now upgraded in terms of both deployment via Ansible and usage via [Open Virtual Network](http://www.ovn.org/).
+
+While still experimental, OVN-DPDK in oVirt is now available in both the stable and master oVirt branches.
```
Comment (gregsheremeta) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: Since this is for end users, change "stable" to what exact version the user can use.

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 9 above)
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: As in non existent device? yes, we'll explode, but I think that's a given? What else should be expected from specifying non existing devices? A fail silently scenario should be explicitly documented, not the other way around imo

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 12 above)
Comment (gregsheremeta) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: It would be nice to have a UI for this. I'm currently working on a project to expose Ansible roles / playbooks to oVirt UI:

https://trello.com/b/8UmL1LBD/rhv-ovirt-extensions

Let me know if you'd like help.

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 13 above)
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 12 above)
Comment (gregsheremeta) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: i added you to the board.

Here's the project doc: https://docs.google.com/document/d/1n4VlosdPnIZL81wx0sEIrkAfBehuV5DfOU7eH9isWT0/edit

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 9 above)
Comment (dankenigsberg) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: oh, now I see that I was misguided by the double (or triple, or quadruple) negation. Consider turning

non-DPDK compatible drivers may be set; NICs not using DPDK compatible drivers won't be configured for OVS-DPDK usage and won't have their CPU isolated in multiple NUMA environments.

into

Non-DPDK compatible drivers may be set; However, NICs with such drivers will not be configured for OVS-DPDK usage and will not have their CPU isolated in multiple NUMA environments.

Comment (dankenigsberg) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: I'm not sure that this is precise: "Empty strings represent kernel's default"

"The empty string represents the default Linux driver for the specified device"

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+    * IOMMU: ```/sys/devices/virtual/iommu/<DMAR device>/devices/<PCI address>``` should exist. DMAR devices are typically annotated by dmar0, dmar1, and so forth.
+    * hugepages: ```grep Huge /proc/meminfo``` should reflect the desired state as defined in the Ansible playbook (i.e. `Hugepagesize` and `HugePages_Total`)
+    * CPU partitioning: based on the devices that are to be used with a DPDK compatible driver, CPUs should be partitioned separately between each NUMA node. Refer to `lscpu` to see live CPU NUMA separation information, e.g.: `NUMA node1 CPU(s):     8-15`
+
+    Note: NICs using userspace drivers do not appear in the kernel; `ip` commands won't list devices using userspace drivers. in oVirt, such devices appear as `dpdk0`, `dpdk1`, and so forth.
+4) Create an oVirt network ("phys-net-demo") on your data center and attach it to your cluster.
+![](https://www.ovirt.org/images/create_phys_net.png)
+5) Attach "phys-net-demo" to DPDK NICs accross your cluster.
+![](https://www.ovirt.org/images/set_phys_net.png)
```
Comment (dankenigsberg) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: it is horrible that we dump all images in a single directory. At least, make sure that all your dpdk images have a dpdk_ prefix, so it would be clearer to understand where they belong.

Comment (dankenigsberg) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: @leongold shouldn't you add an "apology" about SELinux, and point to the relevant bug?

**Comment by dankenigsberg** [comment]
@moshesheena, please review.

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+
+### Ansible DPDK host setup
+
+Host configuration for DPDK usage is now automated using Ansible. This primarly includes:
+* Hugepages configuration -- hugepage size and quantity in the kernel.
+* CPU partitioning.
+* Binding NICs to userspace drivers. 
+* OVS-DPDK related configuration (initialization, socket memory, pmd thread core binding, etc).
+
+The role is installed via Ansible galaxy:
```
Comment (MosheSheena) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: Worth mentioning that the playbook can run either on the engine or on the host.

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+    bind_drivers: true
+    set_ovs_dpdk: false
+  roles:
+    - ovirt-ansible-dpdk-setup
+```
+The role is controlled by 3 boolean variables (all set to `true` by default) and a dictionary of devices and their drivers:
+* `configure_kernel` -- determines whether the kernel should be configured for DPDK usage (hugepages, CPU partitioning). WARNING: When set to `true` it is very likely to trigger reboot of the host, unless all required configuration is already pre-set.
+* `bind_drivers` -- determines whether the role should bind devices to their specified drivers.
+* `set_ovs_dpdk` -- determines whether the role should initialize and configure OVS for DPDK usage.
+* `pci_drivers` -- a dictionary of the PCI addresses of network interface cards as keys and drivers as values. The empty string represents the default Linux driver for the specified device. Non-DPDK compatible drivers may be set; However, NICs with such drivers will not be configured for OVS-DPDK usage and will not have their CPU isolated in multiple NUMA environments.
```
Comment (MosheSheena) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: Maybe add that the pci addresses can be obtained using
lshw -class network -businfo

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+      "0000:02:00.2": "igb"
+      "0000:02:00.3": ""
+    configure_kernel: true
+    bind_drivers: true
+    set_ovs_dpdk: false
+  roles:
+    - ovirt-ansible-dpdk-setup
+```
+The role is controlled by 3 boolean variables (all set to `true` by default) and a dictionary of devices and their drivers:
+* `configure_kernel` -- determines whether the kernel should be configured for DPDK usage (hugepages, CPU partitioning). WARNING: When set to `true` it is very likely to trigger reboot of the host, unless all required configuration is already pre-set.
```
Comment (MosheSheena) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: WARNING should be in bold

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+
+For additional information, refer to the role's [repository](https://github.com/ovirt/ovirt-ansible-dpdk-setup/).
+
+### OVN Integration
+DPDK in oVirt now leverages the capabilities of OVN in the form of an OVN localnet. This allows for seamless connectivity across OVN networks, benefiting from OVN's software defined routers, switches, security groups, and ACL's.
+
+For more information about OVN localnet integration in oVirt, refer to oVirt's [Provider Physical Network RFE](https://ovirt.org/develop/release-management/features/network/provider-physical-network/).
+
+# Usage in oVirt
+1) Install `oVirt.dpdk-setup` Ansible role via ansible-galaxy:
```
Comment (MosheSheena) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: Using the engine vm console - I think it's more explicit

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+    ```
+    # ansible-galaxy install oVirt.dpdk-setup
+    ```
+2) Execute the role as described above.
```
Comment (MosheSheena) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: Maybe rephrase as: Run the playbook after filling all the playbook's parameters.
                                 Make sure the host name you provide in "hosts:" is known in /etc/ansible/hosts file, e.g:
                                 <in playbook>hosts: nari04 
                                 <in hosts file added lines> [nari04]
                                                                            10.10.10.10

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 17 above)
Comment (mmirecki) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: trigger reboot -> trigger a reboot

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 16 above)
Comment (mmirecki) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: The empty -> An empty

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 20]
```diff
@@ -0,0 +1,90 @@
...
+4) Create an oVirt network ("phys-net-demo") on your data center and attach it to your cluster.
+![](https://www.ovirt.org/images/ovn-dpdk/create_phys_net.png)
+5) Attach "phys-net-demo" to DPDK NICs accross your cluster.
+![](https://www.ovirt.org/images/ovn-dpdk/set_phys_net.png)
+6) Create an external OVN network ("ext-net-demo"), setting "phys-net-demo" as its physical network. This configuration between "ext-net-demo" and "phys-net-demo" will enable traffic between "phys-net-demo" (which DPDK's physical port is part of) and the rest of the ports present in OVN's "ext-net-demo" network.
+![](https://www.ovirt.org/images/ovn-dpdk/create_ext_net.png)
+7) "ext-net-demo" vNic's may now be added to virtual machines. These vNics now share L2 connectivity with other remote ports in "ext-net-demo" via DPDK's NIC (and other local ports internally).
+![](https://www.ovirt.org/images/ovn-dpdk/ext_net_demo.png)
+8) Temporarily, SElinux needs to be set to permissivie. This is due to an [open bug](https://bugzilla.redhat.com/1598435) where SElinux blocks the creation of a QEMU socket. Once the bug is resolved, this step should be skipped.
```
Comment (MosheSheena) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: Please correct "permissivie" to permissive
Maybe add that to perform a "one the spot" change use:
setenforce 0
But that will only last until the next reboot
or edit /etc/selinux/config for a permanent change since the kernel boots using this config file

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 21]
```diff
@@ -0,0 +1,90 @@
...
+9) Host to VM packets are transmitted and received on buffers allocated on shared hugepages memory. Virtual machines using DPDK based NIC's need the following custom properties set:
+
```
Comment (MosheSheena) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: Maybe add that vm should be launched using 'run once', and that if the `hugepages_shared` property doesn't show up then use the two commands to config and restart engine service

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 22]
```diff
@@ -0,0 +1,90 @@
...
+Additionally, there are several optional performance related variables:
+* `pmd_threads_count` -- determines the number of PMD threads per NIC (default: `1`).
+* `nr_2mb_hugepages` -- determines the number of 2MB hugepages, if 2MB hugepages are to be used (default: `1024`).
+* `nr_1gb_hugepages` -- determines the number of 1GB hugepages, if 1GB hugepages are to be used (default: `4`).
+* `use_1gb_hugepages` -- determines whether 1GB hugepages should be used, where 1GB hugepages are supported (default: `true`).
+
+For additional information, refer to the role's [repository](https://github.com/ovirt/ovirt-ansible-dpdk-setup/).
+
+### OVN Integration
+DPDK in oVirt now leverages the capabilities of OVN in the form of an OVN localnet. This allows for seamless connectivity across OVN networks, benefiting from OVN's software defined routers, switches, security groups, and ACL's.
```
Comment (mmirecki) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: maybe:
This allows for seamless -> This allows seamless

Comment (mmirecki) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: benefiting from OVN's -> taking advantage of OVN's

File: source/blog/2018-07-19-ovn-dpdk.html.md [hunk 23]
```diff
@@ -0,0 +1,90 @@
...
+
+For more information about OVN localnet integration in oVirt, refer to oVirt's [Provider Physical Network RFE](https://ovirt.org/develop/release-management/features/network/provider-physical-network/).
+
+# Usage in oVirt
+1) Install `oVirt.dpdk-setup` Ansible role via ansible-galaxy:
+    ```
+    # ansible-galaxy install oVirt.dpdk-setup
+    ```
+2) Execute the role as described above.
+3) Unless the host was already configured, the host will restart for kernel changes to be applied. Note that after reboot the following values are changed (If they are not, you may have incompatible hardware, or you should report a bug on ovirt@users.org):
```
Comment (mmirecki) [inline source/blog/2018-07-19-ovn-dpdk.html.md]: (If -> (if

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 19 above)
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: I think that providing an example playbook is enough, and that instructing how to use ansible hosts is too ansible specific

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 15 above)
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Redundant IMO, I also think that running the playbook on the host is impossible in case a reboot is required?

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 18 above)
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: You may install this wherever, it doesn't have to be Engine

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 21 above)
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done (engine restart instruction is already present)

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 17 above)
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 16 above)
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 23 above)
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 22 above)
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Isn't benefiting better here? take advantage to me borders on a negative trait. Is there a grammatical preference for take advantage I'm not aware of?

Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 16 above)
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 17 above)
Comment (leongold) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Done

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 15 above)
Comment (MosheSheena) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Ok i'm convinced

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 18 above)
Comment (MosheSheena) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Ok I was unaware and do agree with you

File: source/blog/2018-07-19-ovn-dpdk.html.md (hunk 19 above)
Comment (MosheSheena) [reply source/blog/2018-07-19-ovn-dpdk.html.md]: Your'e right - using the above should require prior knowledge of ansible. I agree this blog should stick to it's point and not expand on other topics related

**Comment by dankenigsberg** [comment]
@gregsheremeta thanks. However, the publication date ended up wrong. Can you fix it? It sound be July 29

**Comment by gregsheremeta** [comment]
Sure, #1695
